### PR TITLE
Make sidecar more resilient to failure

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

**Comment by cole-h** [comment]
I'd like to get https://github.com/DeterminateSystems/nixos-vault-service/pull/19 in first, because I feel that one is more important, and merging this first would cause a merge conflict (well, merging either of them will do that, but it'll be easier to deal with here).

File: module/implementation.nix [hunk 1]
```diff
@@ -63,6 +63,9 @@ let
 
     in
     {
+      requires = [ "network.target" ];
+      after = [ "network.target" ];
```
[the remark this answers is not in the document]
Comment (grahamc) [reply module/implementation.nix:67]: Let's keep it for now and we might be able to disable it later. I think the vast majority of of the deployed cases will need network.

**Review by grahamc — APPROVED** [review]

File: module/implementation.nix [hunk 2]
```diff
@@ -71,6 +74,8 @@ let
 
       unitConfig = {
         # BindsTo = [ fullServiceName ];
+        StartLimitIntervalSec = 200;
+        StartLimitBurst = 6;
```
[the remark this answers is not in the document]
Comment (cole-h) [reply module/implementation.nix:79]: https://github.com/DeterminateSystems/nixos-vault-service/issues/24

**Comment by cole-h** [comment]
Well, GitHub doesn't want to actually post the check status it seems... but CI succeeded: https://github.com/DeterminateSystems/nixos-vault-service/actions/runs/2035908605.

**Comment by cole-h** [comment]
Well, GitHub doesn't want to actually attach the check status to this PR it seems... but CI succeeded: https://github.com/DeterminateSystems/nixos-vault-service/actions/runs/2035908605.

**Pull request merged**